File: java/src/main/java/org/sparky/model/Filter.java
package org.sparky.model;

import org.sparky.model.exceptions.InvalidColumnNameException;
import org.sparky.model.exceptions.InvalidKeyException;
import sun.reflect.generics.reflectiveObjects.NotImplementedException;

/**
 * Created by lloyd on 26/10/2015.
 */
public class Filter {

    private Rule valueKey;

    public Filter(Rule valueKey) {
        this.valueKey = valueKey;
    }

    public boolean hasChanged(Configuration root) throws InvalidKeyException {
        return true;
    }

    public String getValue(Configuration root) throws InvalidKeyException {
        return valueKey.getValue(root, null);
    }
}

File: net/Sparky/Model/Filter.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Sparky.Model
{
    public class Filter
    {
        private Rule _valueKey;

        public Filter(Rule key)
        {
            _valueKey = key;
        }

        public bool HasChanged(IConfiguration root)
        {
            return true;
        }

        public string Value(IConfiguration root)
        {
            return _valueKey.GetValue(root, null);
        }
    }
}
